modules/core/src/main/scala/lucuma/odb/api/model/SpectroscopyRequirements.scala

```diff
@@ -0,0 +1,133 @@
+// Copyright (c) 2016-2021 Association of Universities for Research in Astronomy, Inc. (AURA)
+// For license information see LICENSE or https://opensource.org/licenses/BSD-3-Clause
+
+package lucuma.odb.api.model
+
+import cats.Eq
+import cats.syntax.all._
+import eu.timepit.refined._
+import eu.timepit.refined.numeric.Positive
+import eu.timepit.refined.cats._
+import eu.timepit.refined.types.numeric.PosBigDecimal
+import eu.timepit.refined.types.numeric.PosInt
+import io.circe.Decoder
+import io.circe.generic.semiauto._
+import io.circe.refined._
+import lucuma.odb.api.model.FocalPlane
+import lucuma.odb.api.model.SpectroscopyCapabilities
+import lucuma.core.math.Angle
+import lucuma.core.util.Display
+import lucuma.core.math.Wavelength
+import lucuma.core.util.Enumerated
+
+final case class SpectroscopyScienceRequirements(
+  wavelength:       Option[Wavelength],
+  resolution:       Option[PosInt],
+  signalToNoise:    Option[PosBigDecimal],
+  signalToNoiseAt:  Option[Wavelength],
+  wavelengthRange:  Option[Wavelength],
+  focalPlane:       Option[FocalPlane],
+  focalPlaneAngle:  Option[Angle],
+  capabilities:     Option[SpectroscopyCapabilities]
+)
+
+object SpectroscopyScienceRequirements {
+  val Default = SpectroscopyScienceRequirements(None, refineMV[Positive](1).some, None, None, None, None, None, None)
+
+  implicit val eqSpectroscopyConfigurationOptions: Eq[SpectroscopyScienceRequirements] = Eq.by(x =>
+    (x.wavelength,
+     x.resolution,
+     x.signalToNoise,
+     x.signalToNoiseAt,
+     x.wavelengthRange,
+     x.focalPlane,
+     x.focalPlaneAngle,
+     x.capabilities
+    )
+  )
+}
+
+object SpectroscopyScienceRequirementsModel {
+  final case class Input(
+    wavelength:       Option[WavelengthModel.Input],
+    resolution:       Option[PosInt],
+    signalToNoise:    Option[PosBigDecimal],
+    signalToNoiseAt:  Option[WavelengthModel.Input],
+    wavelengthRange:  Option[WavelengthModel.Input],
+    focalPlane:       Option[FocalPlane],
+    focalPlaneAngle:  Option[FocalPlaneAngleInput],
+    capabilities:     Option[SpectroscopyCapabilities]
+  ) {
+    val create: ValidatedInput[SpectroscopyScienceRequirements] =
+      (wavelength.traverse(_.toWavelength("wavelength")),
+       signalToNoiseAt.traverse(_.toWavelength("signalToNoiseAt")),
+       wavelengthRange.traverse(_.toWavelength("wavelengthRange")),
+       focalPlaneAngle.traverse(_.toAngle)
+      ).mapN { (cw, signalToNoiseAt, wavelengthRange, focalPlaneAngle) =>
+        SpectroscopyScienceRequirements(cw, resolution, signalToNoise, signalToNoiseAt, wavelengthRange, focalPlane, focalPlaneAngle, capabilities)
+      }
+    val edit: ValidatedInput[SpectroscopyScienceRequirements] =
+      create
```

Comment:
When editing, you don't want to provide a way to remove an option once specified?

Reply:
This was one of my questions. I'm not sure how this goes? Is the idea to set `null` to remove an option?
If so, yes we want to have that option